web/src/lib/stores/assets-store.svelte.ts

```diff
@@ -498,15 +492,50 @@ export class AssetBucket {
   }
 
   findAssetAbsolutePosition(assetId: string) {
+    this.store.clearDeferredLayout(this);
     for (const group of this.dateGroups) {
       const intersectingAsset = group.intersetingAssets.find((asset) => asset.id === assetId);
       if (intersectingAsset) {
-        return this.top + group.top + intersectingAsset.position!.top + this.store.headerHeight;
+        if (!intersectingAsset.position) {
+          console.warn('No position for asset');
+          break;
+        }
+        return this.top + group.top + intersectingAsset.position.top + this.store.headerHeight;
       }
     }
     return -1;
   }
 
+  *assets() {
```

Comment:
What is the `*` for this?

Reply:
This is a generator function. It allows the function to pause and resume context via the yield keyword. It always returns an iterable, and every call to next() on the iterable will resume the function until the yield keyword. 

The purpose of this function is to treat the assets that are found under the store, under the bucket, under the date groups as a single iterable, effectively concatenating all the arrays together. 

This could have been performed as a series of flatMaps, but that would have eagerly copied every object into a new array. 

The generator treats all assets that are scattered all over the place as a single iterable, but without creating any new arrays, and short circuiting of the iteration finds what it’s looking for early.